diff --git a/src/modules/gpu/img-bxm/linux/rogue_km/build/linux/defs.mk b/src/modules/gpu/img-bxm/linux/rogue_km/build/linux/defs.mk
index 40bae004..4e289c15 100644
--- a/src/modules/gpu/img-bxm/linux/rogue_km/build/linux/defs.mk
+++ b/src/modules/gpu/img-bxm/linux/rogue_km/build/linux/defs.mk
@@ -78,7 +78,7 @@ include $$(MAKE_TOP)/moduledefs/$$(MODULE_ARCH).mk
 include $$(MAKE_TOP)/moduledefs_libs.mk
 ifneq ($$(wildcard $$(MAKE_TOP)/modules/$$(strip $$($$(THIS_MODULE)_type)).mk),)
 include $$(MAKE_TOP)/modules/$$(strip $$($$(THIS_MODULE)_type)).mk
-.SECONDARY: $$(MODULE_INTERMEDIATES_DIR)
+.NOTINTERMEDIATE: $$(MODULE_INTERMEDIATES_DIR)
 $$(MODULE_INTERMEDIATES_DIR):
 	$$(make-directory)
 MODULE_CLEAN_TARGETS += $$(MODULE_INTERMEDIATES_DIR)
diff --git a/src/modules/gpu/img-bxm/linux/rogue_km/build/linux/kbuild/Makefile.template b/src/modules/gpu/img-bxm/linux/rogue_km/build/linux/kbuild/Makefile.template
index 9435c15a..75987636 100644
--- a/src/modules/gpu/img-bxm/linux/rogue_km/build/linux/kbuild/Makefile.template
+++ b/src/modules/gpu/img-bxm/linux/rogue_km/build/linux/kbuild/Makefile.template
@@ -54,7 +54,7 @@
 
 include $(OUT)/config_kernel.mk
 
-.SECONDARY:
+.NOTINTERMEDIATE:
 
 define symlink-source-file
 @if [ ! -e $(dir $@) ]; then mkdir -p $(dir $@); fi
diff --git a/src/modules/gpu/img-bxm/linux/rogue_km/build/linux/toplevel.mk b/src/modules/gpu/img-bxm/linux/rogue_km/build/linux/toplevel.mk
index 113de13d..fc98c908 100644
--- a/src/modules/gpu/img-bxm/linux/rogue_km/build/linux/toplevel.mk
+++ b/src/modules/gpu/img-bxm/linux/rogue_km/build/linux/toplevel.mk
@@ -216,13 +216,13 @@ endif
 # them.
 OUT_SUBDIRS := $(addprefix $(RELATIVE_OUT)/,$(TARGET_ALL_ARCH)) \
  $(TARGET_NEUTRAL_OUT) $(DOCS_OUT) $(if $(HOST_PRIMARY_ARCH),$(sort $(HOST_OUT) $(HOST_32BIT_OUT)))
-.SECONDARY: $(OUT_SUBDIRS)
+.NOTINTERMEDIATE: $(OUT_SUBDIRS)
 $(OUT_SUBDIRS):
 	$(make-directory)
 
 ifeq ($(PVR_CREATE_TARGET_PRIMARY_OUT_LINK),1)
 TARGET_PRIMARY_OUT_LINK := $(RELATIVE_OUT)/target_primary
-.SECONDARY: $(TARGET_PRIMARY_OUT_LINK)
+.NOTINTERMEDIATE: $(TARGET_PRIMARY_OUT_LINK)
 $(TARGET_PRIMARY_OUT_LINK):
 	$(LN) $(TARGET_PRIMARY_ARCH) $@
 
@@ -275,7 +275,7 @@ include $(MAKE_TOP)/commands.mk
 
 BUILD_CACHES=$(sort $(LWS_BUILD_CACHES))
 ifneq ($(BUILD_CACHES),)
-.SECONDARY: $(BUILD_CACHES)
+.NOTINTERMEDIATE: $(BUILD_CACHES)
 $(BUILD_CACHES):
 	$(make-directory)
 endif
